glfs.c

```diff
@@ -636,34 +616,15 @@ static int tcmu_glfs_write(struct tcmu_device *dev,
 	return SAM_STAT_TASK_SET_FULL;
 }
 
-static int tcmu_glfs_get_image_size(struct tcmu_device *dev,
-                                    uint64_t new_size)
-{
-	struct glfs_state *gfsp = tcmu_get_dev_private(dev);
-	struct stat st;
-	int ret;
-
-	ret = glfs_lstat(gfsp->fs, gfsp->hosts->path, &st);
-	if (ret) {
-		tcmu_dev_err(dev, "glfs_lstat failed: %m\n");
-		return ret;
-	}
-
-	if (st.st_size != new_size) {
-		tcmu_dev_err(dev, "Mismatched sizes. glfs image size %lld. Requested new size %" PRIu64 ".\n",
-		                  (long long) st.st_size, new_size);
-		return -EINVAL;
-	}
-
-	return 0;
-}
-
 static int tcmu_glfs_reconfig(struct tcmu_device *dev,
                               struct tcmulib_cfg_info *cfg)
 {
 	switch (cfg->type) {
 	case TCMULIB_CFG_DEV_SIZE:
-		return tcmu_glfs_get_image_size(dev, cfg->data.dev_size);
+		/*
+		 * Let glusterfs tools handle size checks
+		 */
+		return 0;
```

Comment:
What would be the benefit in having the size check done by GLFS tools and removing the reconfig of the size here? To me it just looks like you are losing a feature?

Reply:
Thanks for asking this @bgly :-)

1. When the glusterfs tools are already doing a size check before calling targetcli create/reconf then why do same stuff in tcmu-runner, feel it is just a overhead.

2. It will be difficult for the tools to switch between growing and shrinking the sizes while resize failures.

-> say if gluster tool has to perform resize (increase size) operation for a block configured on 3 nodes (HA 3). What we have to do in general is grow the size of the file and  issue targetcli /backstores/user:glfs/block set attribute dev_size=$NEWSIZE

-> But if the targetcli  command fails in one of the nodes, we will have to rollback, which involves shrinking of the size back to old size (dangerous)

But if there is no size check in tcmu-runner, first we run targetcli commands, only if all the HA number of nodes  return success then increase the size with confidence.

This is the flexibility that we get by having the size check at a layer above :-)

Reply:
Thanks for the explanation! It looks good to me then.